pulp_rpm/app/viewsets.py

```diff
@@ -135,13 +135,13 @@ def sync(self, request, pk):
         if repository.retain_package_versions > 0 and mirror:
             raise DRFValidationError("Cannot use 'retain_package_versions' with mirror-mode sync")
 
-        result = enqueue_with_reservation(
+        result = dispatch(
             tasks.synchronize,
             [repository, remote],
             kwargs={
                 "mirror": mirror,
-                "remote_pk": remote.pk,
-                "repository_pk": repository.pk,
+                "remote_pk": str(remote.pk),
+                "repository_pk": str(repository.pk),
```

Comment:
I know we're trying to make these not necessary, but I don't think that's currently released, and 3 of them isn't a huge deal.